Libraries/LibCrypto/NumberTheory/ModularFunctions.h

```diff
@@ -329,15 +333,34 @@ static UnsignedBigInteger random_number(const UnsignedBigInteger& min, const Uns
 
 static bool is_probably_prime(const UnsignedBigInteger& p)
 {
-    if (p == 2 || p == 3 || p == 5)
-        return true;
-    if (p < 49)
+    // Is it a small number?
+    if (p < 49) {
+        u32 p_value = p.words()[0];
+        // Is it a very small prime?
+        if (p_value == 2 || p_value == 3 || p_value == 5 || p_value == 7)
+            return true;
+        // Is it the multiple of a very small prime?
+        if (p_value % 2 == 0 || p_value % 3 == 0 || p_value % 5 == 0 || p_value % 7 == 0)
+            return false;
+        // Then it must be a prime, but not a very small prime, like 37.
         return true;
+    }
 
     Vector<UnsignedBigInteger, 256> tests;
-    UnsignedBigInteger seven { 7 };
-    for (size_t i = 0; i < tests.size(); ++i)
```

Comment:
Schrödinger's primes - prime until observed!

Reply:
Yup, that's actually the thing that made me suspicious in the first place. I thought "Wait, either `Vector` has a horrible API or … oh no. Could it be that it doesn't actually run any tests?!" ^^